crates/sui-indexer-alt/src/handlers/mod.rs

```diff
@@ -0,0 +1,425 @@
+                    // two indexers writing to the same table: They cannot both update the
+                    // watermark. One needs to subordinate to the other (or we need to set the
+                    // watermark to the max of what is currently set and what was about to be
+                    // written).
+
+                    metrics
+                        .total_committer_batches_attempted
+                        .with_label_values(&[H::NAME])
+                        .inc();
+
+                    metrics
+                        .committer_batch_size
+                        .with_label_values(&[H::NAME])
+                        .observe(batch.len() as f64);
+
+                    let guard = metrics
+                        .committer_commit_latency
+                        .with_label_values(&[H::NAME])
+                        .start_timer();
+
+                    match H::commit(&batch, &mut conn).await {
+                        Ok(affected) => {
+                            let elapsed = guard.stop_and_record();
+
+                            metrics
+                                .total_committer_rows_committed
+                                .with_label_values(&[H::NAME])
+                                .inc_by(batch.len() as u64);
+
+                            metrics
+                                .total_committer_rows_affected
+                                .with_label_values(&[H::NAME])
+                                .inc_by(affected as u64);
+
+                            debug!(
+                                pipeline = H::NAME,
+                                elapsed_ms = elapsed * 1000.0,
+                                attempt,
+                                affected,
+                                committed = batch.len(),
+                                pending = pending_rows,
+                                "Wrote batch",
+                            );
+
+                            pending_rows -= batch.len();
+                            attempt = 0;
+
+                            batch.clear();
+                            cool.reset();
+                        }
+
+                        Err(e) => {
+                            let elapsed = guard.stop_and_record();
+
+                            error!(
+                                pipeline = H::NAME,
+                                elapsed_ms = elapsed * 1000.0,
+                                attempt,
+                                committed = batch.len(),
+                                pending = pending_rows,
+                                "Error writing batch: {e}",
+                            );
+
+                            cool.reset_after(RETRY_INTERVAL);
+                            attempt += 1;
+                        }
+                    }
+
+                }
+
+                // If there are enough pending rows, and we've expended the cooldown, reset the
```

Comment:
@lxfind, here is the bit that controls writing things out if you have enough data, but the polling interval hasn't been reached -- we reset the polling interval so that on the next loop around (which immediately follows), the `poll.tick()` will fire.